packages/contact-center/user-state/src/helper.ts

```diff
@@ -1,11 +1,56 @@
-export const useUserState = () => {
+import {useState, useEffect} from "react";
 
+export const useUserState = ({idleCodes, agentId, cc}) => {
 
-  const handleAgentStatus = (event: { target: { value: string; }; }) => {
-  };
+  const [isSettingAgentStatus, setIsSettingAgentStatus] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const [elapsedTime, setElapsedTime] = useState(0);
+  const [currentState, setCurrentState] = useState({});
+
+  useEffect(() => {
+    // Reset the timer whenever the component mounts or the state changes
+    setElapsedTime(0);
+    const timer = setInterval(() => {
+      setElapsedTime(prevTime => prevTime + 1);
+    }, 1000);
+
+    // Cleanup the timer on component unmount
+    return () => clearInterval(timer);
+  }, []);
 
-  const setAgentStatus = () => {
+  const setAgentStatus = (selectedCode) => {
+    const {
+      auxCodeId,
+      state
+    } = {
+      auxCodeId: selectedCode.id,
+      state: selectedCode.name
+    }
+    setIsSettingAgentStatus(true);
+    let oldState = {
+      ...currentState
+    };
+    setCurrentState(selectedCode);
+    const chosenState = state === 'Available' ? 'Available' : 'Idle';
+    cc.setAgentState({state: chosenState, auxCodeId, agentId, lastStateChangeReason: state}).then((response) => {
+      setIsSettingAgentStatus(false);
+      setErrorMessage('');
+      setElapsedTime(0);
+    }).catch(error => {
+      setCurrentState(oldState);
+      setErrorMessage(error.toString());
+      setIsSettingAgentStatus(false);
```

Comment:
This is being set to false irrespective of success or failure here so we can just put it once to false and I see this is only being used for CSS purpose. Can we not use any of other parameters to control the CSS?

Reply:
We can put this in .finally()

Reply:
OPEN ITEM: In the promise we might have a way to figure out if a promise is in the pending state, which will eliminate the use of IsSettingAgentStatus